conformance/lang/expressions/let-expr/let_expr.balt

```diff
@@ -0,0 +1,600 @@
+    int age;
+};
+
+type Student record {|
+    readonly string name;
+    int id?;
+    Address address;
+|};
+
+type Address record {
+    string city;
+    string? country;
+};
+
+function init() {
+    io:println(let table<map<any>> tbl = table [{ id: 13 , name: "Emm", lname: "Dean" }] in tbl); // @output [{"id":13,"name":"Emm","lname":"Dean"}]
+    io:println(() == let table<Foo> key(m) tbl = table [{m: {"AAA": "DDDD"}, age: 31 }]
+        in tbl.add({ m: {"CCC":"EEEE"}, age: 34 })); // @output true
+    io:println(let table<Student> key(name) tbl = table [{name: "Amy", id: 1234, address:
+        {"street": "Main Street", "city": "Colombo", "country": ()}}] in tbl["Amy"]["address"]["country"] == ()); // @output true
+}
+
+Test-Case: output
+Description: Test let-expr with query expressions.
+Labels: let-expr, record-type, array-type, list-constructor-expr
+
+type Department record {|
+   int id;
+   string name;
+|};
+
+type Person record {|
+   int id;
+   string fname;
+   string lname;
+|};
+
+type DeptPerson record {|
+   string fname;
+   string lname;
+   string? dept;
+|};
+
+Person[] personList = [
+    {id: 1, fname: "Alex", lname: "George"}
+];
+
+Department[] deptList = [
+    {id: 1, name:"HR"}
+];
+
+function init() {
+    var l = let Person[] personList = [{id: 1, fname: "Sam", lname: "George"}],
+                   Department[] deptList = [{id: 1, name:"HR"}] in
+                   from var person in personList
+                        join var {id: deptId, name: deptName} in deptList
+                        on person.id equals deptId
+                        select {
+                            fname : person.fname,
+                            lname : person.lname,
+                            dept : deptName
+                        };
+    io:println(l); // @output [{"fname":"Sam","lname":"George","dept":"HR"}]
+}
+
+Test-Case: parser-error
+Description: let-expr with missing tokens.
+Labels: let-expr
+
+function errorFunction() {
+    io:println(let int i in i); // @error missing equal token
+                                // @error missing identifier
+    io:println(let i = 2 in i); // @error unknown type 'i'
+                                // @error missing identifier
+                                // @error undefined symbol 'i'
+    io:println(let int i = 2 i); // @error missing in keyword
+    io:println(let int i = 2 int j = 2 in i + j); // @error missing comma token
+    io:println(let in i); // @error missing let variable declaration
+                          // @error undefined symbol 'i'
+    io:println(let i); // @error unknown type 'i'
+                       // @error missing equal token
+                       // @error missing identifier
+                       // @error missing identifier
+                       // @error missing identifier
+                       // @error missing in keyword
+}
+
+Test-Case: error
+Description: let-expr when variables are declared outside its' scope.
```

Comment:
Not a complete sentence. Please fix wherever applicable.

Reply:
Doesn't reflect what's being tested either.

Reply:
Changed